### DISCUSSION TITLE
Question/answer doesn't work for barriers and tents

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#3945] [General]
Player doesn't receive answer in travelers tent. However, for opening barrier it is not required, To open barrier, player needs only a visited status for a tent. 
![image](https://user-images.githubusercontent.com/55348946/102220104-6ce63a00-3ef1-11eb-940c-88ae297c25b1.png)

**Comment by LeHerosInconnu** [comment]
@Branikolog.

It was really annoying to have to remember the password for the barriers in Heroes 2.

It's a good improvement in fheroes2 that only the condition of having visited the Traveller's Tent can open the Barrier.

Maybe it would be nice to add the "password" in the message window text of the Traveller's Tent and Barrier, but that would just be for fun, without going back to the boring operation of the original game. :)

**Comment by Branikolog** [comment]
I've read somewhere a discussion regarding this topic... Probably here, but couldn't find it with search tool... 


> It was really annoying to have to remember the password for the barriers in Heroes 2.

At the same time, I like the feeling, that I'm doing something special, while entering a password. Something, that depends on my personal skills. So, if I make a mistake - nothing will happen...  How often do we type text in heroes? This is so unusual, and I like it. :)

Also, password can be transferred via not only tent, by event messages, signs, or ally player...  So this expands playing experience. 
So I vote for passwords. )

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> At the same time, I like the feeling, that I'm doing something special, while entering a password. Something, that depends on my personal skills. So, if I make a mistake - nothing will happen... How often do we type text in heroes? This is so unusual, and I like it. :)
> 
> Also, password can be transferred via not only tent, by event messages, signs, or ally player... So this expands playing experience.
> So I vote for passwords. )

Play a XL map over several real days with all the tents and barriers present.

Maybe as an option, let's call it the "masochist" option. :)
But not for me in anyway. :D

**Comment by ihhub** [comment]
A downside of using passwords is inability to enter it if you play a map made in different language which expects answers in that language.

**Comment by ihhub** [comment]
I'm making this topic as a discussion as both solutions have right to exist.

**Comment by Branikolog** [comment]
So, Both operation modes of tents&barriers have some advantages and disadvantages... What if we combine them two into one? Let's say, that after visiting tent a player goes through barrier on default. Otherwise, player can tap a button "if he thinks he knows the secret word" and type in a special box the answer manually.

**Reply by LeHerosInconnu** [reply]
Hello everyone,

> So, Both operation modes of tents&barriers have some advantages and disadvantages... What if we combine them two into one? Let's say, that after visiting tent a player goes through barrier on default. Otherwise, player can tap a button "if he thinks he knows the secret word" and type in a special box the answer manually.

In the original English version of the game, the player must visit the tent and enter the password for the barrier to open.
If the player knows the password but does not visit the tent, the password entered does not open the barrier.

**Reply by Branikolog** [reply]
Humm... My proposal can be abused by players, when playing the same map a couple of times...
Anyway, I like the option to type answers, but just with tent visiting this feature would become more simple, more Heroes3-like...

**Comment by minuspinus** [comment]
It would be great to add the color of the tent from the barrier in the description when visiting the tent.
Sometimes it happens that the tent is hidden by the landscape, and in other cases it would be more obvious.

**Reply by zenseii** [reply]
Could you be more specific. Also, this isn't relevant for the discussion. Feel free to open an issue with a good explanation of the change you want.

**Reply by minuspinus** [reply]
https://github.com/ihhub/fheroes2/issues/9684
